pkg/app/mutate_app.go

```diff
@@ -155,3 +168,46 @@ func (m *Mutator) mutateKubeConfig(ctx context.Context, app v1alpha1.App) ([]mut
 
 	return result, nil
 }
+
+func (m *Mutator) mutateLabels(ctx context.Context, app v1alpha1.App) ([]mutator.PatchOperation, error) {
+	var result []mutator.PatchOperation
+
+	// Set app label if there is no app label present.
+	if key.AppKubernetesNameLabel(app) == "" && key.AppLabel(app) == "" {
+		result = append(result, mutator.PatchAdd(fmt.Sprintf("/metadata/labels/%s", replaceToEscape(label.AppKubernetesName)), key.AppName(app)))
+	}
+
+	// Set version label to be the same as the chart-operator app CR. This
+	// is the version we need and means we don't need to check for a cluster CR.
```

Comment:
Like the comment says this is so the admission controller is not coupled with cluster or release CRs.

Once we run an app-operator per cluster I think we can remove the version label. As we will watch all CRs in the tenant cluster namespace in the control plane cluster.